lib/kitchen/element_base.rb

```diff
@@ -504,6 +504,7 @@ def parent
 
     # returns previous element
     # skips double indentations that the nokigiri sometimes picks up
+    # also skips text belonging to the parent node
     # nil if there's no previous sibling
     #
     def previous
```

Comment:
Switch the nokigiri#previous call to a nokigiri#previous_element?

Reply:
Actually, it _would_ skip text, or at least was doing so when I tried to use it in the #preceded_by_text method. previous_element only returns the next previous element-- no text. I'm not sure why this behavior was missed the first time around, but here we are.